client/src/banners/maybe-banner.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { Suspense, lazy } from "react";
+
+import { useUserData } from "../user-context";
+
+// We may or may not load any active banner. But if there's a small chance
+// that we might, it's best practice to not have to lazy-load the CSS
+// because lazy loading the CSS will put itself as blocking on the critical
+// rendering. So, basically, if there's a >0% chance that we might load
+// <ActiveBanner>, at least the CSS will be ready.
+import "../kumastyles/components/banners/base.scss";
+import "../kumastyles/components/banners/developer-needs.scss";
+import "../kumastyles/components/banners/mdn-subscriptions.scss";
+
+import { DEVELOPER_NEEDS_ID, SUBSCRIPTION_ID } from "./ids";
+
+const ActiveBanner = lazy(() => import("./active-banner"));
```

Comment:
So, this is essentially now the template for displaying an active banner should there be one, correct? If so, I wonder whether we should name this just `Banner` or perhaps `BannerTmpl`.

Then we can name `MaybeBanner`, `ActiveBanner` as it will return the active banner if there is one, and the template will render it. Thoughts @peterbe?

Reply:
I don't like the name "template". Because it's not what it makes. It makes banners. Not templates. Inside it, it has all the necessary texts and URLs to render fully formed banners components.

Reply:
Sounds like a factory pattern (see „Gang of Four”) then.